docs/developer/decision-records/2022-02-07-micrometer-metrics/README.md

```diff
@@ -0,0 +1,132 @@
+
+### Prometheus
+
+We adapted sample 04.3 to deploy an open-source Prometheus server:
+
+```yaml
+  prometheus:
+    image: prom/prometheus:v2.30.3
+    container_name: prometheus
+    volumes:
+      - ./prometheus/:/etc/prometheus/
+    ports:
+      - 9090:9090
+```
+
+The server is the available at [http://localhost:9090](http://localhost:9090).
+
+We added an environment variable for the OpenTelemetry agent to automatically expose a Prometheus endpoint on default port 9464.
+
+```yaml
+ # Env var for consumer and producer
+ OTEL_METRICS_EXPORTER: prometheus
+```
+
+We created a suitable file at `prometheus/prometheus.yml` for the Prometheus server to scrape those endpoints.
+
+```yaml
+global:
+  scrape_interval: 10s
+  scrape_timeout: 10s
+
+scrape_configs:
+  - job_name: services
+    static_configs:
+      - targets:
+          - 'consumer:9464'
+        labels:
+          service: 'Consumer'
+      - targets:
+          - 'provider:9464'
+        labels:
+          service: 'Provider'
+```
+
+### Collecting System and JVM metrics
+
+At startup, we run the following code:
+
+```java
+        var registry = Metrics.globalRegistry;
+        new ClassLoaderMetrics().bindTo(registry);
+        new JvmMemoryMetrics().bindTo(registry);
+        new JvmGcMetrics().bindTo(registry);
+        new ProcessorMetrics().bindTo(registry);
+        new JvmThreadMetrics().bindTo(registry);
+```
+
+This tracks among others the number of live threads:
+
+![jvm-metrics](jvm-metrics.png)
+
+### Collecting OkHttp metrics
+
+OkHttp is an HTTP client library used in EDC to perform outgoing REST calls. Micrometer [natively integrates with OkHttp](https://micrometer.io/docs/ref/okhttpclient).
+
+An `OkHttpClient` instance is globally confirmed in the `:core:base` module. We instrument the client with a metrics listener:
+
+```java
+ OkHttpClient client = new OkHttpClient.Builder()
+                .eventListener(OkHttpMetricsEventListener.builder(Metrics.globalRegistry, "okhttp.requests").build())
+```
+
+This yields count and duration histogram metrics, tagged by target URL.
+
+![okhttp-metrics](okhttp-metrics.png)
+
+### Collecting Jetty metrics
+
+Jetty is the HTTP server and servlet container used in EDC to expose HTTP endpoints. [Micrometer natively integrates with Jetty](https://github.com/micrometer-metrics/micrometer/tree/main/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jetty), although the integration is not documented. We referred to [Spring Boot code](https://github.com/spring-projects/spring-boot/blob/ce6b12a02480f77ead612834123ba640509649c9/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/web/jetty/JettyConnectionMetricsBinder.java#L49) to understand how to achieve integration:
```

Comment:
As you mention `although the integration is not documented.`, so do you see a possibility to improve this open source documentation as part of our sharing phase ? A potential backlog item.

Same for `Jersey` below.